addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/handler/HomematicBridgeHandler.java

```diff
@@ -0,0 +1,307 @@
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.homematic.discovery.HomematicDeviceDiscoveryService;
+import org.openhab.binding.homematic.internal.common.HomematicConfig;
+import org.openhab.binding.homematic.internal.communicator.HomematicGateway;
+import org.openhab.binding.homematic.internal.communicator.HomematicGatewayFactory;
+import org.openhab.binding.homematic.internal.communicator.HomematicGatewayListener;
+import org.openhab.binding.homematic.internal.misc.HomematicClientException;
+import org.openhab.binding.homematic.internal.misc.LocalNetworkInterface;
+import org.openhab.binding.homematic.internal.model.HmDatapoint;
+import org.openhab.binding.homematic.internal.model.HmDevice;
+import org.openhab.binding.homematic.type.HomematicTypeGenerator;
+import org.openhab.binding.homematic.type.UidUtils;
+import org.osgi.framework.ServiceRegistration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link HomematicBridgeHandler} is the handler for a Homematic gateway and connects it to the framework.
+ *
+ * @author Gerhard Riegler - Initial contribution
+ */
+public class HomematicBridgeHandler extends BaseBridgeHandler implements HomematicGatewayListener {
+    private static final Logger logger = LoggerFactory.getLogger(HomematicBridgeHandler.class);
+    private static final long REINITIALIZE_DELAY_SECONDS = 10;
+    private static SimplePortPool portPool = new SimplePortPool();
+
+    private HomematicConfig config;
+    private HomematicGateway gateway;
+    private HomematicTypeGenerator typeGenerator;
+
+    private HomematicDeviceDiscoveryService discoveryService;
+    private ServiceRegistration<?> discoveryServiceRegistration;
+
+    public HomematicBridgeHandler(Bridge bridge, HomematicTypeGenerator typeGenerator) {
+        super(bridge);
+        this.typeGenerator = typeGenerator;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialize() {
+        try {
+            String id = getThing().getUID().getId();
+            config = createHomematicConfig();
+
+            gateway = HomematicGatewayFactory.createGateway(id, config, this);
+            gateway.initialize();
+
+            registerDeviceDiscoveryService();
+            scheduler.submit(new Runnable() {
+
+                @Override
+                public void run() {
+                    discoveryService.waitForScanFinishing();
+                    updateStatus(ThingStatus.ONLINE);
+                    for (Thing hmThing : getThing().getThings()) {
+                        hmThing.getHandler().thingUpdated(hmThing);
+                    }
+                }
+            });
+
+        } catch (IOException ex) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ex.getMessage());
+            logger.error(ex.getMessage(), ex);
```

Comment:
please remove any error logging after doing a thing status update - the thing status change is logged and that is enough.